idaes/core/scaling/custom_scaler_base.py

```diff
@@ -229,18 +229,47 @@ def get_default_scaling_factor(
         Returns:
             default scaling factor if it exists, else None
         """
-        try:
-            return self.default_scaling_factors[component.local_name]
-        except KeyError:
-            # Might be indexed, see if parent component has default scaling
-            parent = component.parent_component()
-            try:
-                return self.default_scaling_factors[parent.local_name]
-            except KeyError:
-                # No default scaling found, give up
-                pass
+        blk = component.parent_block()
+        comp_default = None
+        parent_default = None
+        # We're not just returning self.default_scaling_factors[component.local_name]
+        # because the component finder has additional logic to handle, e.g., spaces
+        # separating indices for elements of an indexed component. We want the user
+        # to be able to provide either "var[a,b]" or "var[a, b]" (with a space following
+        # the comma) and still find the right component
+
+        # If the user provides both "var[a,b]" and "var[a, b]", then whichever is
+        # encountered first when iterating through the keys will be returned. This
+        # behavior is not ideal. TODO dictionary validation
+
+        # Iterating through every key is certainly not the most efficient way to go
+        # about this look-up process, but it's also probably not going to be the
+        # rate limiting step, especially considering these default dictionaries
+        # should be relatively short.
+
+        # Locking attribute creation context prevents build-on-demand properties
+        # from getting triggered through this lookup.
+        if hasattr(blk, "_lock_attribute_creation"):  # pylint: disable=protected-access
+            lock_attribute_creation_orig = (
+                blk._lock_attribute_creation  # pylint: disable=protected-access
+            )
+            blk._lock_attribute_creation = True  # pylint: disable=protected-access
+        for key in self.default_scaling_factors:
+            comp2 = blk.find_component(key)
+            if comp2 is component:
+                comp_default = self.default_scaling_factors[key]
```

Comment:
```suggestion
            if comp2 is component:
                comp_default = self.default_scaling_factors[key]
                break
```
In the large comment above you mention that if multiple matching string names appear for a component in `default_scaling_factors` then the first value encountered when iterating will be returned. I think you need a `break` here to get that behavior.